tests/OnlineSales.Tests/BaseTest.cs

```diff
@@ -109,22 +130,35 @@ protected async Task<HttpResponseMessage> GetTest(string url, HttpStatusCode exp
             return null;
         }
     }
+
+    protected async Task<Stream?> GetImageTest(string url, HttpStatusCode expectedCode = HttpStatusCode.OK, string authToken = "Success")
+    {
+        var response = await GetTest(url, expectedCode, authToken);
 
-    protected async Task<string> PostTest(string url, object payload, HttpStatusCode expectedCode = HttpStatusCode.Created, string authToken = "Success")
-    {        
-        var response = await Request(HttpMethod.Post, url, payload, authToken);
-
-        response.StatusCode.Should().Be(expectedCode);
-
-        var location = string.Empty;
+        var content = await response.Content.ReadAsStreamAsync();
 
-        if (expectedCode == HttpStatusCode.Created)
+        if (expectedCode == HttpStatusCode.OK)
         {
-            location = response.Headers?.Location?.LocalPath ?? string.Empty;
-            location.Should().StartWith(url);
+            return content;
         }
+        else
+        {
+            return null;
+        }
+    }
+
+    protected async Task<string> PostTest(string url, TestImage payload, HttpStatusCode expectedCode = HttpStatusCode.Created, string authToken = "Success")
```

Comment:
This method should be moved away from BaseClass. Please put it to ImagesTests.cs as it it very specific to that particular test class. We do not need it in any other test cases.